src/com/google/javascript/jscomp/CollapseProperties.java

```diff
@@ -696,6 +700,16 @@ private void collapseDeclarationOfNameAndDescendants(Name n, String alias) {
 
     // Handle this name first so that nested object literals get unrolled.
     if (n.canCollapse()) {
+
+      // Enum properties are always collapsed. Warn when @nocollapse
+      // is used on an enum property
+      if (n.docInfo != null && n.docInfo.isNoCollapse() &&
+          n.parent != null && n.parent.type == Name.Type.OBJECTLIT &&
+          n.parent.docInfo != null && n.parent.docInfo.hasEnumParameterType()) {
```

Comment:
No, but the enumerated type can be an object, eg,

    /** @enum { { a: { b: number}} } */
    var e = {
      KEY1: { a: { /** @nocollapse */ b: 123}},
      KEY2: { a: { b: 456}}
    };

A more realistic usecase (and then one we'd like to prevent) is when people use enums as namespaces. They declare a bunch of "static" properties on the enum name after the definition. These properties can have any type. Sth like:

    /** @enum */
    var e = { A: 1, B: 2 };
    /** @type { c: { d: number } } */
    e.name1 = { c: { /** @nocollapse */ d: 123 } };